### PR TITLE
Add Roslyn project walker

### LINKED ISSUE #4344: Implement C#/VB.NET project walker
### Description

When migrating, it would be useful to identify the set of  C#/VB.NET projects in the solution, if any.
We do not need to visit any other project types, or any other "special" Solution Explorer folders.

Options:
1) use the existing-but-ugly `IProjectHelper` methods to walk the `SolutionExplorer` hierarchy and filter out the projects/folders we are not interested in, or 
2) walk the Roslyn workspace, which will only return the projects we care about.

Option(2) seems like a simpler option. In addition, having the Roslyn `Project` objects will allow us to check whether the project references any additional files or not.

TODO
- [x] add Roslyn project walker
- [x] add tests

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #4344

### CODE REVIEW

File: src/ConnectedMode/Migration/RoslynProjectProvider.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+using Microsoft.VisualStudio.LanguageServices;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    internal interface IRoslynProjectProvider
+    {
+        /// <summary>
+        /// Returns the list of Roslyn (C# and VB.NET) projects, if any
+        /// </summary>
+        IReadOnlyList<Project> Get();
```
Comment (duncanp-sonar) [inline src/ConnectedMode/Migration/RoslynProjectProvider.cs:34]: Simple helper to return a list of the Roslyn (i.e. C# and VB.NET projects) in the current solution.

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithoutStrongNames.txt [hunk 2]
```diff
@@ -155,8 +155,10 @@ Relative path: 'SonarLint.VisualStudio.ConnectedMode.dll'
 Referenced assemblies:
 - 'LibGit2Sharp, Version=0.26.0.0, Culture=neutral, PublicKeyToken=7cbde695407f0333'
 - 'Microsoft.Alm.Authentication, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null'
+- 'Microsoft.CodeAnalysis.Workspaces, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithoutStrongNames.txt:158]: The `VSWorkspace` requires a couple of additional assemblies. These ship with VS so we don't need to add them to the VSIX..

File: src/ConnectedMode/Migration/IVisualStudioWorkspace.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    /// <summary>
+    /// We can't mock of subclass VisualStudioWorkspace because it has a private constructor.
+    /// The wrapper allows us to test the methods we use.
+    /// </summary>
+    internal interface IVisualStudioWorkspace
+    {
+        Solution CurrentSolution { get; }
+        IVsHierarchy GetHierarchy(ProjectId projectId);
+    }
+
```
Comment (bigfluffycookie) [inline src/ConnectedMode/Migration/IVisualStudioWorkspace.cs:37]: space

**Review by bigfluffycookie — APPROVED** [review]
Looks good! One space too much in a file (sorry for being annoying about those)

File: src/ConnectedMode/Migration/IVisualStudioWorkspace.cs (hunk 3 above)
Comment (duncanp-sonar) [reply src/ConnectedMode/Migration/IVisualStudioWorkspace.cs:37]: For me, an extra line blank line isn't a big problem as it doesn't really affect readability. Missing whitespace is more of a problem.

**Pull request merged**